src/main/drivers/dshot_bb.c

```diff
@@ -0,0 +1,697 @@
+        }
+    }
+
+    bbmotor->protocolControl.value = value;
+
+    uint16_t packet = prepareDshotPacket(&bbmotor->protocolControl);
+
+    bbPort_t *bbPort = bbmotor->bbPort;
+
+#ifdef USE_DSHOT_TELEMETRY
+    if (useDshotTelemetry) {
+        bbOutputDataSet(bbPort->portOutputBuffer, bbmotor->pinIndex, packet, BBSHOT_INVERTED);
+    } else
+#endif
+    {
+        bbOutputDataSet(bbPort->portOutputBuffer, bbmotor->pinIndex, packet, BBSHOT_NONINVERTED);
+    }
+}
+
+static void bbWrite(uint8_t motorIndex, float value)
+{
+    bbWriteInt(motorIndex, value);
+}
+
+static void bbUpdateComplete(void)
+{
+    // If there is a dshot command loaded up, time it correctly with motor update
+
+    if (!dshotCommandQueueEmpty()) {
+        if (!dshotCommandOutputIsEnabled(bbDevice.count)) {
+            return;
+        }
+    }
+
+#ifdef USE_DSHOT_TELEMETRY
+    for (int i = 0; i < usedMotorPorts; i++) {
+        bbPort_t *bbPort = &bbPorts[i];
+
+        if (useDshotTelemetry) {
+            if (bbPort->direction == BBSHOT_DIRECTION_INPUT) {
+                bbPort->inputActive = false;
+                bbSwitchToOutput(bbPort);
+            }
+
+
+            bbDMA_Cmd(bbPort, ENABLE);
+        }
+    }
+#endif
+
+    for (int i = 0; i < usedMotorPacers; i++) {
+        bbPacer_t *bbPacer = &bbPacers[i];
+        bbTIM_DMACmd(bbPacer->tim, bbPacer->dmaSources, ENABLE);
+    }
+}
+
+static bool bbEnableMotors(void)
+{
+    for (int i = 0; i < motorCount; i++) {
+        IOConfigGPIO(bbMotors[i].io, bbMotors[i].iocfg);
+    }
+    return true;
+}
+
+static void bbDisableMotors(void)
+{
+    return;
+}
+
+static void bbShutdown(void)
+{
+    return;
+}
+
+static bool bbIsMotorEnabled(uint8_t index)
+{
+    return bbMotors[index].enabled;
+}
+
+static const motorDevConfig_t *lastMotorConfig;
+
+static void bbPostInit()
+{
+    bbFindPacerTimer();
+
+    for (int motorIndex = 0; motorIndex < MAX_SUPPORTED_MOTORS && motorIndex < motorCount; motorIndex++) {
+
+        if (!bbMotorConfig(bbMotors[motorIndex].io, motorIndex, lastMotorConfig->motorPwmProtocol, bbMotors[motorIndex].output)) {
+            return NULL;
+        }
+
+
+        bbMotors[motorIndex].enabled = true;
+
+        // Fill in motors structure for 4way access (XXX Should be refactored)
+
+        motors[motorIndex].enabled = true;
+    }
+}
+
+static motorVTable_t bbVTable = {
+    .postInit = bbPostInit,
+    .enable = bbEnableMotors,
+    .disable = bbDisableMotors,
+    .isMotorEnabled = bbIsMotorEnabled,
+    .updateStart = bbUpdateStart,
+    .write = bbWrite,
+    .writeInt = bbWriteInt,
+    .updateComplete = bbUpdateComplete,
+    .convertExternalToMotor = dshotConvertFromExternal,
+    .convertMotorToExternal = dshotConvertToExternal,
+    .shutdown = bbShutdown,
+};
+
+
+motorDevice_t *bbshotDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8_t count, bool useUnsyncedPwm)
+{
+    UNUSED(idlePulse);
+    UNUSED(useUnsyncedPwm);
+
+    dbgPinInit();
```

Comment:
What is this used for?

Reply:
I defer to @jflyper

Reply:
These provide minimal latency timing marker output used for optimizing raw I/O codes. We can strip them now, but I would like to keep them for a while, until all platform support is done (H7 and possibly G4).

Reply:
They seem to be generically useful for debugging timing issues - I think they should be extracted into separate files, in a separate pull request.

Or at least they need to be commented properly, so that if we forget to remove them in the end, the next person editing the code will not have to spend hours guessing and testing if this functionality is actually required for bitbanged Dshot to work.